test/Window/Event.test.cpp

```diff
@@ -16,7 +16,253 @@ TEST_CASE("[Window] sf::Event")
+        const auto& mouseMoved = *event.getIf<sf::Event::MouseMoved>();
+        CHECK(mouseMoved.position == sf::Vector2i(4, 2));
+
+        event = sf::Event::MouseEntered{};
+        CHECK(event.is<sf::Event::MouseEntered>());
+        CHECK(event.getIf<sf::Event::MouseEntered>());
+
+        event = sf::Event::MouseLeft{};
+        CHECK(event.is<sf::Event::MouseLeft>());
+        CHECK(event.getIf<sf::Event::MouseLeft>());
+
+        event = sf::Event::JoystickButtonPressed{100, 200};
+        CHECK(event.is<sf::Event::JoystickButtonPressed>());
+        CHECK(event.getIf<sf::Event::JoystickButtonPressed>());
+        const auto& joystickButtonPressed = *event.getIf<sf::Event::JoystickButtonPressed>();
+        CHECK(joystickButtonPressed.joystickId == 100);
+        CHECK(joystickButtonPressed.button == 200);
+
+        event = sf::Event::JoystickButtonReleased{300, 400};
+        CHECK(event.is<sf::Event::JoystickButtonReleased>());
+        CHECK(event.getIf<sf::Event::JoystickButtonReleased>());
+        const auto& joystickButtonReleased = *event.getIf<sf::Event::JoystickButtonReleased>();
+        CHECK(joystickButtonReleased.joystickId == 300);
+        CHECK(joystickButtonReleased.button == 400);
+
+        event = sf::Event::JoystickMoved{300, sf::Joystick::Axis::Z, 1.23f};
+        CHECK(event.is<sf::Event::JoystickMoved>());
+        CHECK(event.getIf<sf::Event::JoystickMoved>());
+        const auto& joystickMoved = *event.getIf<sf::Event::JoystickMoved>();
+        CHECK(joystickMoved.joystickId == 300);
+        CHECK(joystickMoved.axis == sf::Joystick::Axis::Z);
+        CHECK(joystickMoved.position == 1.23f);
+
+        event = sf::Event::JoystickConnected{42};
+        CHECK(event.is<sf::Event::JoystickConnected>());
+        CHECK(event.getIf<sf::Event::JoystickConnected>());
+        const auto& joystickConnected = *event.getIf<sf::Event::JoystickConnected>();
+        CHECK(joystickConnected.joystickId == 42);
+
+        event = sf::Event::JoystickDisconnected{43};
+        CHECK(event.is<sf::Event::JoystickDisconnected>());
+        CHECK(event.getIf<sf::Event::JoystickDisconnected>());
+        const auto& joystickDisconnected = *event.getIf<sf::Event::JoystickDisconnected>();
+        CHECK(joystickDisconnected.joystickId == 43);
+
+        event = sf::Event::TouchBegan{99, {98, 97}};
+        CHECK(event.is<sf::Event::TouchBegan>());
+        CHECK(event.getIf<sf::Event::TouchBegan>());
+        const auto& touchBegan = *event.getIf<sf::Event::TouchBegan>();
+        CHECK(touchBegan.finger == 99);
+        CHECK(touchBegan.position == sf::Vector2i(98, 97));
+
+        event = sf::Event::TouchMoved{96, {95, 94}};
+        CHECK(event.is<sf::Event::TouchMoved>());
+        CHECK(event.getIf<sf::Event::TouchMoved>());
+        const auto& touchMoved = *event.getIf<sf::Event::TouchMoved>();
+        CHECK(touchMoved.finger == 96);
+        CHECK(touchBegan.position == sf::Vector2i(95, 94));
```

Comment:
```suggestion
        CHECK(touchMoved.position == sf::Vector2i(95, 94));
```
funny that it worked nonetheless because being at the same address.